dags/process_elife_zip_dag.py

```diff
@@ -103,39 +131,19 @@ def extract_archived_files_to_bucket(**context) -> str:
             raise FileNotFoundError(error_message)
 
     # pass article cloud bucket key to next task
-    return '%s/%s' % (folder_name, article_name)
+    return '%s/%s' % (prefix, article_name)
 
 
-def convert_tiff_images_in_expanded_bucket_to_jpeg_images(**context) -> None:
+def convert_tiff_images_in_expanded_bucket_to_jpeg_images(**context) -> List[str]:
     zip_file_name = get_file_name_passed_to_dag_run_conf_file(context)
     prefix = zip_file_name.replace('.zip', '/')
 
-    s3 = get_aws_connection('s3')
-
-    for key in list_bucket_keys_iter(Bucket=DESTINATION_BUCKET, Prefix=prefix):
-        if key.endswith('.tif'):
-            with TemporaryFile(dir=TEMP_DIRECTORY) as temp_tiff_file:
-                s3.download_fileobj(
-                    Bucket=DESTINATION_BUCKET,
-                    Key=key,
-                    Fileobj=temp_tiff_file
-                )
-                temp_tiff_file.seek(0)
-
-                logger.info('converting %s to jpeg', key)
+    tiffs = {key
+             for key in list_bucket_keys_iter(Bucket=DESTINATION_BUCKET, Prefix=prefix)
+             if key.endswith('.tif')}
 
-                temp_jpeg = BytesIO()
-                with Image(file=temp_tiff_file) as img:
-                    img.format = 'jpeg'
-                    img.save(file=temp_jpeg)
-
-                key = re.sub(r'\.\w+$', '.jpg', key)
-                s3.put_object(
-                    Bucket=DESTINATION_BUCKET,
-                    Key=key,
-                    Body=temp_jpeg.getvalue()
-                )
-                logger.info('%s uploaded to %s',key, DESTINATION_BUCKET)
+    with multiprocessing.Pool() as p:
+        return p.map(convert_image_in_s3_to_jpeg, tiffs, chunksize=3)
```

Comment:
is `3` the size of the pool? It has implications for memory as the more concurrent processes are there the more memory needs to be allocated for each of them.

Reply:
no. The pool size is equivalent to the number or cores. `chunksize` refers to the number of items to pass to a process before spawning a new process. This means that if `tiffs` has <= 3 items only one process will be spawned.

Reply:
Ok, "chunk" means "batch" size here